cmake/alpakaCommon.cmake

```diff
@@ -279,18 +279,22 @@ if(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE OR A
 
     if(OpenMP_CXX_FOUND)
         if(ALPAKA_ACC_ANY_BT_OMP5_ENABLED)
-            if(OpenMP_CXX_VERSION VERSION_LESS 4.0)
-                message(FATAL_ERROR "ALPAKA_ACC_ANY_BT_OMP5_ENABLE requires compiler support for OpenMP at least 4.0, 5.0 is recommended.")
-            elseif(OpenMP_CXX_VERSION VERSION_LESS 5.0)
-                message(WARNING "OpenMP < 5.0, for ALPAKA_ACC_ANY_BT_OMP5_ENABLE 5.0 is recommended.")
+            if(OpenMP_CXX_VERSION VERSION_LESS 5.0)
+                message(FATAL_ERROR "ALPAKA_ACC_ANY_BT_OMP5_ENABLE requires compiler support for OpenMP 5.0.")
+
+                if((${CMAKE_CXX_COMPILER_ID} STREQUAL "AppleClang") AND (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 12.0.5))
+                    message(FATAL_ERROR "The OpenMP 5.0 back-end requires Xcode 12.5 or later")
+                elseif((${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang") AND (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 11.0))
+                    message(FATAL_ERROR "The OpenMP 5.0 back-end requires clang 11.0 or later")
+                endif()
```

Comment:
Have we agreed on dropping support for OpenMP 4+? I am fine, but I think we did not have a discussion.

Reply:
Well, it's either this or no support for `#pragma omp flush acq_rel` and having to resort to older OpenMP standards. But admittedly I just assumed that we'd be okay with this since we're also okay with raising the required compiler version quite significantly.